app/components/animated_number/index.tsx

```diff
@@ -24,44 +24,68 @@ const usePrevious = (value: number) => {
 
 const AnimatedNumber = ({
     animateToNumber,
-    animationDuration,
+    animationDuration = 1400,
     fontStyle,
     easing = 1.2,
 }: Props) => {
-    const prevNumber = usePrevious(animateToNumber);
+    const previousNumber = usePrevious(animateToNumber);
+    const [numberHeight, setNumberHeight] = useState(0);
+
     const animateToNumberString = String(Math.abs(animateToNumber));
-    const prevNumberString = String(Math.abs(prevNumber));
 
-    const [numberHeight, setNumberHeight] = React.useState(0);
+    const previousNumberString = useMemo(() => {
+        // comparing previousNumber and animateToNumber gets trickier when the number of digits changes and the length differs.
+        // By padding the previousNumber with 0s, or slicing the previousNumber to match the length of animateToNumber
+        // we can compare them digit by digit
+        const _previousNumberString = String(Math.abs(previousNumber)).padStart(
+            animateToNumberString.length,
+            '0',
+        );
+
+        if (_previousNumberString.length > animateToNumberString.length) {
```

Comment:
@rahimrahman, in which case this code will execute, as we are adding padding to the string with 0, this means the `_previousNumberString` and `anumateToNumberString` lengths will always be equal.

Reply:
@Rajat-Dabade When it goes from `1000` to `999` or something extreme like `100000001` to `1`.  The AnimateNumber doesn't really know what's being fed into it, so we have to be ready for any scenarios.

Take `1000` to `999` as an example.  The padStart() only adds leading 0 to the length of animateToNumberString.length to _previousNumberString (not to animateToNumberString).  In this case, it sliced `1000` to match the length of `999` thus becoming `000`.  Then this component knows to animate from `0` to `9` in all 3 digits.

I actually left a comment.  If it wasn't clear, maybe we need to tweak it.